ted2zim/multi/scraper.py

```diff
@@ -0,0 +1,330 @@
+            download_link("https://www.ted.com/topics/combo?models=Talks").text
+        )
+        if mode == "topic":
+            return topic_list
+
+        # mode is playlist
+        s3_url_with_credentials = None
+        for arg in self.extra_args:
+            if "--optimization-cache" in arg:
+                # handle both = seperated and space seperated values
+                if len(arg) > 21:
+                    s3_url_with_credentials = arg[21:]
+                else:
+                    s3_url_with_credentials = self.extra_args[
+                        self.extra_args.index(arg) + 1
+                    ]
+
+        if s3_url_with_credentials:
+            s3_storage = KiwixStorage(s3_url_with_credentials)
+            if not s3_storage.check_credentials(
+                list_buckets=True, bucket=True, write=True, read=True, failsafe=True
+            ):
+                logger.error("S3 credential check failed. Continuing without S3")
+                return self.download_playlist_list_from_site(topic_list)
+            self.output_dir.mkdir(parents=True, exist_ok=True)
+            playlist_list = self.download_playlist_list_from_cache(s3_storage)
+            if not playlist_list:
+                playlist_list = self.download_playlist_list_from_site(topic_list)
+                self.upload_playlist_list_to_cache(playlist_list, s3_storage)
+            return playlist_list
+        return self.download_playlist_list_from_site(topic_list)
+
+    def run(self):
+        def log_run_result(success, process):
+            if success:
+                logger.info(".. OK")
+            else:
+                logger.error(".. ERROR. Printing scraper output and exiting.")
+                logger.error(process.stdout)
+                return process.returncode
+
+        logger.info(f"starting {NAME}-multi scraper")
+
+        self.fetch_metadata()
+
+        if self.topics:
+            if self.topics == ["all"]:
+                self.topics = [
+                    topic["value"] for topic in self.get_list_of_all(mode="topic")
+                ]
+            if self.indiv_zims:
+                for topic in self.topics:
+                    logger.info(f"Executing ted2zim for topic {topic}")
+                    success, process = self.run_indiv_zim_mode(topic, mode="topic")
+                    log_run_result(success, process)
+
+            else:
+                logger.info("Falling back to ted2zim for topic(s)")
+                if not self.handle_single_zim(mode="topic"):
+                    logger.error("ted2zim Failed")
+
+        if self.playlists:
+            if self.playlists == ["all"]:
+                self.playlists = [
+                    str(playlist["id"])
+                    for playlist in self.get_list_of_all(mode="playlist")
+                ]
+            # automatically set self.indiv_zims if not already set and multiple playlists
+            if len(self.playlists) > 1 and not self.indiv_zims:
```

Comment:
Good idea but I think it should fail instead and before that (in entrypoint?). What about the name for instance?

Reply:
I added it to allow creation of aims if someone wants ZIMs of multiple topics as one zim but also has added multiple playlists to the command